kpatch-build/create-diff-object.c

```diff
@@ -3056,18 +3056,23 @@ static void kpatch_create_mcount_sections(struct kpatch_elf *kelf)
 		memcpy(newdata, sym->sec->data->d_buf, sym->sec->data->d_size);
 		sym->sec->data->d_buf = newdata;
 		insn = newdata;
-		if (insn[0] != 0xf)
-			ERROR("%s: unexpected instruction at the start of the function",
-			      sym->name);
-		insn[0] = 0xe8;
-		insn[1] = 0;
-		insn[2] = 0;
-		insn[3] = 0;
-		insn[4] = 0;
 
 		rela = list_first_entry(&sym->sec->rela->relas, struct rela,
 					list);
-		rela->type = R_X86_64_PC32;
+		if (!(insn[0] == 0xe8 && insn[1] == 0 && insn[2] == 0 &&
+		    insn[3] == 0 && insn[4] == 0) ||
+		    rela->type != R_X86_64_PC32) {
```

Comment:
Can the above check for opcode/rela type replaced with `if (rela->type == R_X86_64_NONE)`

Reply:
Hm, yes. I think right now this should work as well and definitely does look better. It will be a bit less strict though, but I guess it is worth it. If there won't be any objections to this I'll change it to this, thank you.